components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py

```diff
@@ -0,0 +1,168 @@
+"""SageMaker component for transform."""
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Dict
+
+from batch_transform.src.sagemaker_transform_spec import (
+    SageMakerTransformSpec,
+    SageMakerTransformInputs,
+    SageMakerTransformOutputs,
+)
+from common.sagemaker_component import (
+    SageMakerComponent,
+    ComponentMetadata,
+    SageMakerJobStatus,
+)
+
+
+@ComponentMetadata(
+    name="SageMaker - Batch Transformation",
+    description="Batch Transformation Jobs in SageMaker",
+    spec=SageMakerTransformSpec,
+)
+class SageMakerTransformComponent(SageMakerComponent):
+    """SageMaker component for transform."""
+
+    def Do(self, spec: SageMakerTransformSpec):
+        self._transform_job_name = (
+            spec.inputs.job_name
+            if spec.inputs.job_name
+            else "BatchTransform"
```

Comment:
Why do we do this instead of using the auto generated job names (like the other components)?

Reply:
I think we should use auto generated names, lets change it

Reply:
What do you mean by "pass job name"?